raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+            partner_locksroot: typing.Locksroot
+    ):
+        """ Settle the channel.
+
+        Raises:
+            ChannelBusyError: If the channel is busy with another operation
+        """
+        log.info(
+            'settle called',
+            token_network=pex(self.address),
+            node=pex(self.node_address),
+            partner=pex(partner),
+            transferred_amount=transferred_amount,
+            locked_amount=locked_amount,
+            locksroot=encode_hex(locksroot),
+            partner_transferred_amount=partner_transferred_amount,
+            partner_locked_amount=partner_locked_amount,
+            partner_locksroot=encode_hex(partner_locksroot),
+        )
+
+        if not self.channel_operations_lock.acquire(blocking=False):
+            raise ChannelBusyError(
+                f'Channel with address {self.address} is '
+                f'busy with another ongoing operation'
+            )
+
+        with releasing(self.channel_operations_lock):
+            transaction_hash = estimate_and_transact(
+                self.proxy,
+                'settleChannel',
+                self.node_address,
+                transferred_amount,
+                locked_amount,
+                locksroot,
+                partner,
+                partner_transferred_amount,
+                partner_locked_amount,
+                partner_locksroot,
+            )
+
+            self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+            receipt_or_none = check_transaction_threw(self.client, transaction_hash)
+            if receipt_or_none:
+                log.info(
+                    'settle failed',
+                    token_network=pex(self.address),
+                    node=pex(self.node_address),
+                    partner=pex(partner),
+                    transferred_amount=transferred_amount,
+                    locked_amount=locked_amount,
+                    locksroot=encode_hex(locksroot),
+                    partner_transferred_amount=partner_transferred_amount,
+                    partner_locked_amount=partner_locked_amount,
+                    partner_locksroot=encode_hex(partner_locksroot),
+                )
+                self._check_exists()
+                raise TransactionThrew('Settle', receipt_or_none)
+
+            log.info(
+                'settle successful',
+                token_network=pex(self.address),
+                node=pex(self.node_address),
+                partner=pex(partner),
+                transferred_amount=transferred_amount,
+                locked_amount=locked_amount,
+                locksroot=encode_hex(locksroot),
+                partner_transferred_amount=partner_transferred_amount,
+                partner_locked_amount=partner_locked_amount,
+                partner_locksroot=encode_hex(partner_locksroot),
+            )
+
+    def events_filter(
+            self,
+            topics: Optional[List],
```

Comment:
So decided to keep the topics argument? It's not used at all anywhere. Why not just remove it?

Reply:
Not sure if it's not going to be used when changing to getting the state from events. I would leave it here for now.